src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,126 @@
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Book book = null;
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                book = mapResultToBook(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get a book by id " + id, e);
+        }
+        return Optional.ofNullable(book);
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String sql = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book book = mapResultToBook(resultSet);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't fetch books", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException(
+                        "Expected to update at least one row, but updated 0 rows for book: "
+                                + book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update a book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String sql = "DELETE FROM books WHERE id = ?";
+        int affectedRows;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            affectedRows = statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete a book with id: " + id, e);
+        }
+        return affectedRows > 0;
+    }
+
+    private Book mapResultToBook(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getLong("id");
```

Comment:
as it was mentioned in the common mistakes
```suggestion
        Long id = resultSet.getObject("id", Long.class);
```